applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h

```diff
@@ -314,33 +315,42 @@ class EigenfrequencyResponseFunctionUtility
                         gradient_contribution[coord_dir_i] += gradient_prefactors[i] * inner_prod(eigenvectors_of_element[i] , aux_vector) * mWeightingFactors[i];
                     }
 
-                   node_i.GetInitialPosition()[coord_dir_i] -= mDelta;
+                    node_i.GetInitialPosition()[coord_dir_i] -= mDelta;
+                    node_i.Coordinates()[coord_dir_i] -= mDelta;
                 }
                 noalias(node_i.FastGetSolutionStepValue(SHAPE_SENSITIVITY)) += gradient_contribution;
             }
         }
     }
 
     // --------------------------------------------------------------------------
-    double GetEigenvalue(int eigenfrequency_id)
+    double GetEigenvalue(const int& eigenfrequency_id)
```

Comment:
why by reference?

Reply:
Thought it makes sense to pass by reference if possible to avoid the copy. No other reason

Reply:
Ok found it on google: Apparently passing by reference does not give a benefit for types like int (maybe even produces overhead).

Reply:
Yep
We usually use it only when it is actually needed
I.e. if you pas an int by reference, then others will assume that sth happens to this int

Other types you should in most cases pass by (const) reference